### ISSUE TITLE
Switch Integration off or on  via Automation

### ISSUE THREAD

**Issue by Wolfi311** [mikey0000/Mammotion-HA#142]
### The problem

It should be possible to deactivate or activate the integration via automation.
Reason:
I switch off charging at night power consumption standby 20W!
But the integration then drains the battery 15% per hour.
Without active integration 10% per 12 hours!
Is there a possibility?
Greetings
Wolfgang

### What Mammotion mower are you using?

Luba 2

### What version of the Mammotion intergration are you using?

0.1.39

### What version of Home Assistant Core are you running?

2024.9.2

**Comment by mikey0000** [comment]
Are you using Bluetooth or WiFi?

But in short yes I can add something for you.

22/09/2024 1:20:55 am Wolfgang ***@***.***>:

**Comment by Wolfi311** [comment]
Wifi 
It would be great if you could start and stop the integration via automation.


Greetings

**Comment by elmr91** [comment]
I had exactly same use case:
Mu Yuka is powered off between mowing sessions (remotely controlled power switch)
Yuka is shutdown itself when idle with no power from station.
Yuka is booting again as soon as power is restored on the charging station.
It worked perfectly this way from weeks.

I tried this integration (Wifi), it works very but keeps Yuka awake : I can confim 15% battery in 2 hours.
It could very useful to be able to pause integration polling by automation.

scenario example:
- when power is restored, enable integration polling
- when switched off, pause integration polling
-

**Comment by mikey0000** [comment]
Problem here is, if its pollong it doesn't turn off, turn on however is possible. Anyway button or switch coming

**Comment by kingunlim** [comment]
i have the same problem....
is there already a solution for this?

best regards and thx for this beautiful integration!

**Comment by mikey0000** [comment]
Please try with the switch in 0.1.58

**Comment by Wolfi311** [comment]
And how do I get the 0.1.58?
So far I have only been offered the 0.1.55 update

**Comment by elmr91** [comment]
I'm running version 0.1.59
I'm trying to test switch: switch.yuka_xxxxxxxxx_turn_updates_on_off

When switch is off, most of entities seems frozen (as expected)
Looking at some "dynamic" entities like L1 or L2 satellites, effect of switch is visible (values frozen)
![image](https://github.com/user-attachments/assets/d1183693-4902-4b8e-be5f-9a8315c58c25)

But strangely, I received some events in the logs while update switch was on off position:
![image](https://github.com/user-attachments/assets/e4c4bc0b-f53c-4a00-9c89-41a7e64a18da)

It seems robot polling is still occurring some way because robot is not going to sleep when swicth is off.

How can I help providing some useful debug logs ?

**Comment by Wolfi311** [comment]
Ok I will test, I have now also installed 0.1.59

**Comment by Wolfi311** [comment]
That doesn't seem to be working yet.
Battery consumption from 100% to 75% in 2 hours, i.e. 25%!
With Power off and Turn Updates off

**Comment by mikey0000** [comment]
OK I'll make it disconnect from MQTT as well, that will definitely stop it.

**Comment by elmr91** [comment]
I upgraded to V 0.1.66 this morning.
I tested to switch off updates. Robot goes to sleep after 10 mn as usual.
I think polling does not occur any more, Idle power consumption should be normal again.
Thanks @mikey0000

**Comment by elmr91** [comment]
Switching updates off is now correctly stopping any polling/activity towards robot.
But switching updates back on does not resume correctly.
Values are not updated anymore after switch off then switch on again. 
I guess the integration does not reconnect correctly.

**Comment by Wolfi311** [comment]
@mikey0000 
@elmr91 

I can confirm elmr91's statement.
I come to the same conclusion


Greetings
Wolfgang

**Comment by mikey0000** [comment]
I'll see if I have some time to look into it. I'm overseas for a couple weeks.

**Comment by Wolfi311** [comment]
One more addition. 
Only a restart of HA reactivates the integration.
Deactivating/activating or reloading is not sufficient.

**Comment by mikey0000** [comment]
Just pushed another release can you give it a go.

**Comment by elmr91** [comment]
> Just pushed another release can you give it a go.

I just updated to 0.1.68, but with no success: it still not reconnecting after switch off/on.
(BTW I faced also #171 problem - I restarted HA while robot was powered down)

**Comment by Wolfi311** [comment]
So it seems to work for me. About after power and update off (12h), power consumption ok.  
Update on ok.
But I haven't tested all the functions yet

**Comment by mikey0000** [comment]
Thanks for the feedback, let me know how you get on this week with it

**Comment by Wolfi311** [comment]
If you press the Update switch in power off when Luba is in sleep mode, the following error messages are displayed.
This is probably normal. ?
Only once did I have to restart HA for the integration to work again.

‘The action switch/turn_on could not be executed. 6205’

‘The action switch/turn_off could not be executed. list.remove(x): x not in list’

**Comment by elmr91** [comment]
I understood why it didn't work for me yesterday: I switched updates to on too quickly after robot power on.
Robot was not ready and integration failed silently (once again similar as #171 I guess)
If I wait 2 mn after power power on for switching updates on, everything works fine.

**Comment by elmr91** [comment]
> If you press the Update switch in power off when Luba is in sleep mode, the following error messages are displayed. This is probably normal. ? Only once did I have to restart HA for the integration to work again.
> 
> ‘The action switch/turn_on could not be executed. 6205’
> 
> ‘The action switch/turn_off could not be executed. list.remove(x): x not in list’

You are right,
This is exactly what happened when trying to switch update on too quickly
![image](https://github.com/user-attachments/assets/fa965c1f-719c-4e5b-b66c-f8330b9992de)

**Comment by Wolfi311** [comment]
This is exactly why I switch the update switch with a 5 min delay after power on (automation)

**Comment by irgendwer112** [comment]
I successfully managed a 80% battery mode with HACS https://spook.boo/. There you can enable/disable integration polling or deactivate integrations.

**Comment by Wolfi311** [comment]
@irgendwer112 
Yes, I know spook.
However, it is not the most elegant solution to end an integration.
Since the integration is under development, you should solve it right away!

Greetings

**Comment by TTLucian** [comment]
For me the switch doesn't work at all with Yuka
```
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:245
integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 19:00:29 (3 occurrences)
Last logged: 19:00:59

[140331473262560] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2761, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2804, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 996, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1068, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/mammotion/switch.py", line 226, in async_turn_off
    await self.entity_description.set_fn(self.coordinator, False)
  File "/config/custom_components/mammotion/coordinator.py", line 102, in set_scheduled_updates
    await device.ble().stop()
  File "/usr/local/lib/python3.12/site-packages/pymammotion/mammotion/devices/mammotion_bluetooth.py", line 121, in stop
    self._ble_sync_task.cancel()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'cancel'
```

**Comment by mikey0000** [comment]
> For me the switch doesn't work at all with Yuka
> 
> ```
> Logger: homeassistant.components.websocket_api.http.connection
> Source: components/websocket_api/commands.py:245
> integration: Home Assistant WebSocket API (documentation, issues)
> First occurred: 19:00:29 (3 occurrences)
> Last logged: 19:00:59
> 
> [140331473262560] Unexpected exception
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
>     response = await hass.services.async_call(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/core.py", line 2761, in async_call
>     response_data = await coro
>                     ^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/core.py", line 2804, in _execute_service
>     return await target(service_call)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 996, in entity_service_call
>     single_response = await _handle_entity_call(
>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1068, in _handle_entity_call
>     result = await task
>              ^^^^^^^^^^
>   File "/config/custom_components/mammotion/switch.py", line 226, in async_turn_off
>     await self.entity_description.set_fn(self.coordinator, False)
>   File "/config/custom_components/mammotion/coordinator.py", line 102, in set_scheduled_updates
>     await device.ble().stop()
>   File "/usr/local/lib/python3.12/site-packages/pymammotion/mammotion/devices/mammotion_bluetooth.py", line 121, in stop
>     self._ble_sync_task.cancel()
>     ^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'cancel'
> ```

Version? Do you have keep Bluetooth connected on?

**Comment by TTLucian** [comment]
Mammotion 0.1.76
HA 2024.10.3
WiFi only, no Bluetooth

**Comment by mikey0000** [comment]
Oh OK, I should be able to have that sorted today

**Comment by mikey0000** [comment]
Should be resolved, can you check again?

**Comment by TTLucian** [comment]
So far so good. No more errors!

**Comment by Wolfi311** [comment]
Turn Updates On/Off works very well for me.
And many other things work better and better. 
Many thanks to you .

I'm currently running automation and scripts with a dashboard for the Luba.  
With the version v0.1.81
I will update to v0.1.88 this week.

Greetings Wolfgang
 
![image](https://github.com/user-attachments/assets/06f2cd80-b6f7-409d-a803-8905f2a4fd72)

**Comment by Wolfi311** [comment]
So the mowing season is over for me, so unfortunately I can no longer test. (Integration)
I cleaned the Luba 2 today and put it away for hibernation!
See you next year.

Greetings and thanks 
Wolfgang

**Comment by mikey0000** [comment]
> So the mowing season is over for me, so unfortunately I can no longer test. (Integration) I cleaned the Luba 2 today and put it away for hibernation! See you next year.
> 
> Greetings and thanks Wolfgang

Thanks, look forward to hearing from you in the next season, I'm sure there will be lots of changes!

**Comment by Wolfi311** [comment]
So the mowing season is over for me, so unfortunately I can no longer test. (Integration)

I cleaned the Luba 2 today and put it away for hibernation!

See you next year.

 

Greetings and thanks 

Wolfgang

 

Von: Michael Arthur ***@***.***> 
Gesendet: Mittwoch, 30. Oktober 2024 09:32
An: mikey0000/Mammotion-HA ***@***.***>
Cc: Wolfgang ***@***.***>; Author ***@***.***>
Betreff: Re: [mikey0000/Mammotion-HA] Switch Integration off or on via Automation (Issue #142)

 

Closed #142 <https://github.com/mikey0000/Mammotion-HA/issues/142>  as completed.
